### PR TITLE
Inc-7055 long term fix: add external ids to message tags

### PR DESCRIPTION
**Pull request by alfrimpong**
[SEV Betterment](https://twilio.enterprise.slack.com/archives/C067L6S9BEY)

This PR adds external id metadata to message tags. This is to be consumed by marketing analytics.

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [x] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/destination-actions/src/destinations/engage/twilio/utils/PhoneMessageSender.ts [hunk 1]
```diff
@@ -48,7 +48,9 @@ export abstract class PhoneMessageSender<Payload extends PhoneMessagePayload> ex
       campaign_name: this.payload.customArgs && this.payload.customArgs['campaign_name'],
       campaign_key: this.payload.customArgs && this.payload.customArgs['campaign_key'],
       user_id: this.payload.customArgs && this.payload.customArgs['user_id'],
-      message_id: this.payload.customArgs && this.payload.customArgs['message_id']
+      message_id: this.payload.customArgs && this.payload.customArgs['message_id'],
+      external_id_type: recepient.type,
+      external_id_value: phone
     }
```
Comment (joelbandi) [inline packages/destination-actions/src/destinations/engage/twilio/utils/PhoneMessageSender.ts:54]: We should also confirm the total tag count we are allowed to send that event stream will return back to us.

Comment (joelbandi) [reply packages/destination-actions/src/destinations/engage/twilio/utils/PhoneMessageSender.ts:54]: Sure!

**Comment by joe-ayoub-segment** [comment]
hi @alfrimpong , @joelbandi thanks for raising this PR. 
Please ping me when you are ready for me to review/approve/deploy. 

Looks like CI is failing ATM. 

The next deploy is tomorrow morning, and the one after that is next Tuesday.

**Review by hocaho — APPROVED** [review]

**Review by joe-ayoub-segment — APPROVED** [review]

File: packages/destination-actions/src/destinations/engage/twilio/utils/PhoneMessageSender.ts (hunk 1 above)
Comment (lpediredla) [reply packages/destination-actions/src/destinations/engage/twilio/utils/PhoneMessageSender.ts:54]: tag size, and also please make sure there are no '%' in the tag body.

Comment (alfrimpong) [reply packages/destination-actions/src/destinations/engage/twilio/utils/PhoneMessageSender.ts:54]: @lpediredla `%` is scrubbed in personas-service.

**Comment by joe-ayoub-segment** [comment]
Hi @alfrimpong this PR has been deployed

**Pull request merged**